Ladybird/Qt/SettingsDialog.cpp

```diff
@@ -35,6 +33,19 @@ SettingsDialog::SettingsDialog(QMainWindow* window)
 
     m_new_tab_page = make<QLineEdit>(this);
     m_new_tab_page->setText(Settings::the()->new_tab_page());
+    QObject::connect(m_new_tab_page, &QLineEdit::textChanged, this, [this] {
```

Comment:
I think we can go with the "save as you go" if it's a bit less aggressive than the `textChanged` signal. Looking at the Qt docs, `editingFinsihed` seems like a good one to use for actually committing the changes to Settings.

https://doc.qt.io/qt-6/qlineedit.html#editingFinished

For setting the border to red we can probably keep using the text changed, or also move it to `editingFinished` ? 

It would be nice to commit settings when the entire window is "left", but when the text box is left seems ok too.

Reply:
I think the validation with red border must stay in `textChanged` so you get immediate feedback. I have moved the save to `editingFinished` as you suggested